cpp/src/gandiva/tests/generate_data.h

```diff
@@ -91,7 +103,7 @@ class FastUtf8DataGenerator : public DataGenerator<std::string> {
     return cur_char_;
   }
 
-  std::random_device random_;
+  Random random_;
   unsigned int max_len_;
   char cur_char_;
 };
```

Comment:
We should merge all of this data generation with the code in `arrow/test-util.h`, probably create `arrow/util/random.h`. cc @pitrou